### ISSUE TITLE
Don't require my <select>'s to include an <option> when ng-options has been specified

### ISSUE THREAD

**Issue by ChrisMBarr** [nikestep/html-angular-validate#9]
If i have the following in an angular controller:

```
$scope.someList = ["one", "two", "three"];
```

And the following in my HTML

```
<select ng-model="vm.selectedDomain"
        ng-options="item as item for item in someList"
        required></select>
```

i get the error

> A “select” element with a “required” attribute, and without a “multiple” attribute, and without a “size” attribute whose value is greater than “1”, must have a child “option” element.

This seems to me like it's wrong, because i shouldn't have to add an `<option>` element since the options are generated dynamically from the list using `ng-options`.  perhaps this error could be bypassed when it notices that `ng-options` is specified?

**Comment by nikestep** [comment]
I've always been hesitant to blanket relax errors. This is really a syntactic wrapper of the validator, not so much a semantic wrapper. I'm happy to look at a pull request if you've got the time to dig in to how to fix this properly, but I'm not going to be able to get to this.

**Comment by ChrisMBarr** [comment]
I tried to fix this with PR #13, but here's what I found out:

Given this input HTML:

```
<select ng-model="vm.selectedDomain" ng-options="item as item for item in someList" required></select>
```

I get this error object back:

```
{ type: 'error',
  lastLine: 13,
  lastColumn: 30,
  firstColumn: 22,
  message: 'A “select” element with a “required” attribute, and without a “multiple” attribute, and without a “size” attribute whose value is greater than “1”, must have a child “option” element.',
  extract: ' required></select>\n\n</di',
  hiliteStart: 10,
  hiliteLength: 9 }
```

I was hoping to be able to check the `extract` property for `ng-options` and the ignore this error if it existed, but it looks like it doesn't return enough relevant HTML to do so. :cry:
